spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/store/jwk/JwkDefinitionSource.java

```diff
@@ -90,7 +100,9 @@ JwkDefinition getDefinitionLoadIfNecessary(String keyId) {
 			return result;
 		}
 		this.jwkDefinitions.clear();
-		this.jwkDefinitions.putAll(loadJwkDefinitions(this.jwkSetUrl));
+		for(URL jwkSetUrl : jwkSetUrls) {
```

Comment:
@jgrandja We already have a race condition between the clear and putAll and was saving that for a separate issue. I pushed 2 more commits to address #1105